regression/scripts/battery/battery_cell_discharge_tests.py

```diff
@@ -0,0 +1,491 @@
+            results = mission.evaluate()  
+            plot_results(results,i,j,battery_chemistry[i], axes1, axes2, axes3, axes4, axes5, axes6, axes7, axes8)  
+    
+
+    legend_font_size = 12                     
+    axes1.set_ylabel('Voltage $(V_{UL}$)')    
+    axes1.set_xlabel('Amp-Hours (A-hr)') 
+    axes1.legend(loc='upper right', prop={'size': legend_font_size})  
+    axes1.set_ylim([2,4.5]) 
+    axes1.set_xlim([0,7])
+    axes2.set_xlabel('Amp-Hours (A-hr)') 
+    axes2.legend(loc='upper right', prop={'size': legend_font_size})  
+    axes2.set_ylim([2,4.5])   
+    axes2.set_xlim([0,7])
+    axes3.set_xlabel('Amp-Hours (A-hr)')
+    axes3.legend(loc='upper right', prop={'size': legend_font_size})  
+    axes3.set_ylim([2,4.5]) 
+    axes3.set_xlim([0,7])
+    axes4.set_xlabel('Amp-Hours (A-hr)') 
+    axes4.legend(loc='upper right', prop={'size': legend_font_size})
+    axes4.set_ylim([2,4.5])    
+    axes4.set_xlim([0,7]) 
+    
+    axes5.set_ylabel(r'Temperature ($\degree$C)')    
+    axes5.set_xlabel('Amp-Hours (A-hr)')        
+    axes5.legend(loc='upper left', prop={'size': legend_font_size})
+    axes5.set_ylim([295,320])
+    axes5.set_xlim([0,7])
+     
+    axes6.set_xlabel('Amp-Hours (A-hr)')     
+    axes6.legend(loc='upper left', prop={'size': legend_font_size})  
+    axes6.set_ylim([295,320])
+    axes6.set_xlim([0,7])
+     
+    axes7.set_xlabel('Amp-Hours (A-hr)')    
+    axes7.legend(loc='upper left', prop={'size': legend_font_size})   
+    axes7.set_ylim([295,320])
+    axes7.set_xlim([0,7])
+     
+    axes8.set_xlabel('Amp-Hours (A-hr)')    
+    axes8.legend(loc='upper left', prop={'size': legend_font_size})      
+    axes8.set_ylim([295,320])
+    axes8.set_xlim([0,7])
+   
+    plt.tight_layout()
+    
+    return 
+
+def plot_results(results,i,j,bat_chem, axes1, axes2, axes3, axes4, axes5, axes6, axes7, axes8): 
+    
+    C_rat  = [0.5,1,2,3]     
+    mark_1 = ['s' ,'s' ,'s' ,'s','s']
+    mark_2 = ['o' ,'o' ,'o' ,'o','o']
+    mark_3 = ['P' ,'P' ,'P' ,'P','P']
+    ls_1   = ['-' ,'-' ,'-' ,'-','-']
+    ls_2   = ['--' ,'--' ,'--' ,'--']
+    ls_3   = [':' ,':' ,':' ,':']
+    lc_1   = ['green' , 'blue' , 'red' , 'orange' ]
+    lc_2   = ['darkgreen', 'darkblue' , 'darkred'  ,'brown']
+    lc_3   = ['limegreen', 'lightblue' , 'pink'  ,'yellow'] 
+    ms = 8
+
+    for segment in results.segments.values():
+        time          = segment.conditions.frames.inertial.time[:,0]/60 
+        volts         = segment.conditions.propulsion.battery_voltage_under_load[:,0]   
+        cell_temp     = segment.conditions.propulsion.battery_cell_temperature[:,0]   
+        Amp_Hrs       = segment.conditions.propulsion.battery_charge_throughput[:,0]    
+        
+        use_amp_hrs = True
+        
+        if use_amp_hrs == True:
+            x_vals = Amp_Hrs
+        else:
+            x_vals = time                   
+        
+        if bat_chem == 'LFP':  
+            if j == 0:
```

Comment:
Instead of this long if-else statement, you can just pass in the label/marker and the two axes you want to plot on to this function; also you don't need any of the elifs for j==x since they are just an index term and you would be passing in the two axes into the function

Reply:
changes made